app/controllers/compute_resources_controller.rb

```diff
@@ -119,13 +119,21 @@ def resource_pools
     end
   end
 
+  def available_images
+    return head(:method_not_allowed) unless @compute_resource.is_a? Foreman::Model::Vmware
```

Comment:
I would like to avoid specific provider filtering here.
We can get this information from the model itself, if we introduce meaningful `nil` result to `#available_images`.

``` ruby
images = @compute_resource.available_images
return head(:method_not_allowed) if images.nil?
#...
```

The result of `#available_images` would be one of the following:
- `nil`: This provider does not support/provide such info
- Non-empty array: List of images that are relevant to this compute resources
- '[]': The provider supports image lists, but the list is empty in this particular case.